client/data/promote-post/use-promote-post-posts-query-paged.ts

```diff
@@ -0,0 +1,77 @@
+import { useInfiniteQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import { SearchOptions } from 'calypso/my-sites/promote-post-i2/components/search-bar';
+
+type BlazablePostsQueryOptions = {
+	page?: number;
+};
+
+const getSearchOptionsQueryParams = ( searchOptions: SearchOptions ) => {
+	let searchQueryParams = '';
+
+	if ( searchOptions.search ) {
+		searchQueryParams += `&title=${ searchOptions.search }`;
+	}
+	if ( searchOptions.filter ) {
+		if ( searchOptions.filter.status && searchOptions.filter.status !== 'all' ) {
+			searchQueryParams += `&status=${ searchOptions.filter.status }`;
+		}
+	}
+	if ( searchOptions.order ) {
+		searchQueryParams += `&order_by=${ searchOptions.order.orderBy }`;
+		searchQueryParams += `&order=${ searchOptions.order.order }`;
+	}
+
+	return searchQueryParams;
+};
+
+async function queryPosts( siteId: number, queryparams: string ) {
```

Comment:
Nice!, now we have the possibility to separate this request from the rest of the WP requests and have a specific endpoint in our package. 

In Jetpack, all requests but the one that the DSP does are redirected to the stats packages (by default), but with this, I can just redirect it to the blaze package without affecting other calls.

Reply:
Would be great if I could get this endpoint tested in jetpack directly. I've also improved the wpcom endpoint to give more compatibility .
If there's something you might need to explore that could ease the jetpack integration, please tell me.

Reply:
> Would be great if I could get this endpoint tested in jetpack directly

You should be able to test the endpoint with this PR, as long as you have Jetpack sites connected to your WordPress.com account at the moment.

Another way would be to add a code snippet to your Jetpack site to make queries to the endpoint from there directly, and log the results to see what you get. Something like this:

```php
/**
 * Test Blaze's Posts endpoint
 *
 * @see D109592-code
 *
 * @return void
 */
function jeherve_test_blaze_endpoints() {
	$site_id  = \Automattic\Jetpack\Connection\Manager::get_site_id();
	$url      = sprintf(
		'/sites/%1$d/blaze/posts?title=%2$s',
		$site_id,
		'A new post about things' // Change for any post or any parameter you want to test on your site.
	);
	$response = \Automattic\Jetpack\Connection\Client::wpcom_json_api_request_as_blog(
		$url,
		'2',
		array( 'method' => 'GET' ),
		null,
		'wpcom'
	);
	$result   = json_decode( wp_remote_retrieve_body( $response ), true );

	error_log( print_r( $result, true ) );
}
add_action( 'init', 'jeherve_test_blaze_endpoints' );
```